src/sharing/index.js

```diff
@@ -0,0 +1,57 @@
+/* @flow strict-local */
+import { NativeModules, DeviceEventEmitter, Platform } from 'react-native';
+import type { Dispatch, SharedData } from '../types';
+import { navigateToSharing } from '../actions';
+
+const { Sharing } = NativeModules;
+
+export const handleReceivedData = async (data: SharedData, dispatch: Dispatch) => {
+  dispatch(navigateToSharing(data));
+};
+
+export const handleInitialShare = async (dispatch: Dispatch) => {
+  const initialSharedData: SharedData | null = await Sharing.getInitialSharedContent();
+  if (initialSharedData !== null) {
+    handleReceivedData(initialSharedData, dispatch);
+  }
+};
+
+export class ShareReceivedListener {
+  dispatch: Dispatch;
+  unsubs: Array<() => void> = [];
+
+  constructor(dispatch: Dispatch) {
+    this.dispatch = dispatch;
+  }
+
+  /** Private. */
+  listen(name: string, handler: (...empty) => void | Promise<void>) {
+    if (Platform.OS === 'android') {
+      const subscription = DeviceEventEmitter.addListener(name, handler);
+      this.unsubs.push(() => subscription.remove());
+    }
+  }
+
+  /** Private. */
+  unlistenAll() {
+    while (this.unsubs.length > 0) {
+      this.unsubs.pop()();
+    }
+  }
+
+  handleShareReceived = (data: SharedData) => {
+    handleReceivedData(data, this.dispatch);
+  };
```

Comment:
This can be inlined, as an arrow function. Unlike in the case where we're passing a callback as a React prop, nothing's going to be looking to see if one value is `===` to a previous one and optimizing if it is -- which is the reason we have to have these separate named handler functions so often elsewhere.

Reply:
Okay, removed `handleReceivedData` and made the calls inline.

Reply:
Sorry, this was ambiguous -- I meant that this function `handleShareReceived` could be inlined :slightly_smiling_face: 

This is fine too, though; IIRC I actually waffled on whether I wanted to inline `handleShareReceived`, or inline `handleReceivedData` at its two call sites including this one.